src/logger/__tests__/MetricScope.test.ts

```diff
@@ -12,7 +12,7 @@ test('async scope executes handler function', async () => {
   // arrange
   let wasInvoked = false;
 
-  const handler = metricScope(metrics => async (evt: any) => {
```

Comment:
This is a good example of the bug that this PR prevents.

This handler requires one parameter (`evt`) of type `any`. 

With the old typing, you were able to call just `handler()` with no parameter. Now, you actually have to put in that parameter and the code will not compile if you have a bug like where you are missing the parameter.